ragna/deploy/_ui/js_utils.py

```diff
@@ -0,0 +1,58 @@
+def preformat(text):
+    """Allows {{key}} to be used for formatting in textcthat already uses
+    curly braces.  First switch this into something else, replace curlies
+    with double curlies, and then switch back to regular braces
+    """
+    text = text.replace("{{", "<<<").replace("}}", ">>>")
+    text = text.replace("{", "{{").replace("}", "}}")
+    text = text.replace("<<<", "{").replace(">>>", "}")
+    return text
+
+
+def redirect_script(remove, append="/", remove_auth_cookie=False):
+    """
+    This function returns a js script to redirect to correct url.
+    :param remove: string to remove from the end of the url
+    :param append: string to append at the end of the url
+    :param remove_auth_cookie: boolean, will clear auth_token cookie when true.
+    :return: string javascript script
+
+    Examples:
+    =========
+
+    # This will remove nothing from the end of the url and will
+    # add auth to it, so /foo/bar/car/ becomes /foo/bar/car/auth
+    >>> redirect_script(remove="", append="auth")
+
+    # This will remove nothing from the end of the url and will
+    # add auth to it, so /foo/bar/car/ becomes /foo/bar/car/logout
+    >>> redirect_script(remove="", append="logout")
+
+    # This will remove "auth" from the end of the url and will add / to it
+    # so /foo/bar/car/auth becomes /foo/bar/car/
+    >>> redirect_script(remove="auth", append="/")
+    """
+    js_script = preformat(
```

Comment:
IIUC, the only reason we need the `preformat` with six `str.replace` is because we want to use `str.format` later and our original text might already have curly braces inside of it, correct? If so, wouldn't it be easier to rely on `re.sub(r"\{\{\w+\}\}")` or whatever other delimiters for the substitution you want to use?

Reply:
So you mean without using `str.format`?